File: rb/features.rb
# NppesNpiRegistry SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module NppesNpiRegistryFeatures
  def self.make_feature(name)
    case name
    when "base"
      NppesNpiRegistryBaseFeature.new
    when "ratelimit"
      NppesNpiRegistryRatelimitFeature.new
    when "retry"
      NppesNpiRegistryRetryFeature.new
    when "test"
      NppesNpiRegistryTestFeature.new
    when "timeout"
      NppesNpiRegistryTimeoutFeature.new
    else
      NppesNpiRegistryBaseFeature.new
    end
  end
end

File: py/features.py
# NppesNpiRegistry SDK feature factory

from feature.base_feature import NppesNpiRegistryBaseFeature
from feature.test_feature import NppesNpiRegistryTestFeature


def _make_feature(name):
    features = {
        "base": lambda: NppesNpiRegistryBaseFeature(),
        "test": lambda: NppesNpiRegistryTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
